### PR TITLE
Move libultra variables to their respective headers

### PR DESCRIPTION
**Pull request by AngheloAlf**
Continuing on the libultra cleanup series, this PR aims to cleanup the `extern`ed variables from `variables.h` and move them to their respective headers.
I also made a few other kinda related changes, like moving `du`, `fu` and the `ROUND` macro to `guint.h`, and move `size_t` and `NULL` to `ultratypes.h`

### CODE REVIEW

File: include/fault.h [hunk 1]
```diff
@@ -103,7 +103,7 @@ typedef struct FaultMgr {
     /* 0x844 */ void* fb;
 } FaultMgr; // size = 0x848
 
-extern FaultMgr gFaultMgr;
-
+// Commented out to try to avoid bss reorder in fault.c
```
Comment (hensldm) [inline include/fault.h:106]: Could you add a TODO so we can find this easier later?

**Review by hensldm — APPROVED** [review]
1 small nit

File: src/libultra/io/vi.c [hunk 2]
```diff
@@ -1,13 +1,13 @@
 #include "global.h"
 
-__OSViContext D_80098060[2] = { 0 };
-__OSViContext* __osViCurr = &D_80098060[0];
-__OSViContext* __osViNext = &D_80098060[1];
+__OSViContext vi[2] = { 0 };
```
Comment (engineer124) [inline src/libultra/io/vi.c:3]: Without this being static, will a generic name `vi` cause any issues with name conflicts?

File: src/libultra/io/controller.c [hunk 3]
```diff
@@ -17,11 +17,11 @@ s32 osContInit(OSMesgQueue* mq, u8* bitpattern, OSContStatus* data) {
     OSTimer mytimer;
     OSMesgQueue timerMesgQueue;
 
-    if (D_80097E40 != 0) {
+    if (__osContinitialized != 0) {
```
Comment (engineer124) [inline src/libultra/io/controller.c:20]:
```suggestion
    if (__osContinitialized) {
```
This seems boolean? Could this also be `__osContInitialized` with the I capitalized?

**Review by engineer124 — APPROVED** [review]

**Review by tom-overton — APPROVED** [review]
I am not a libultra expert, so take this with a grain of salt. However, the changes look good to me.

**Pull request merged**